flink/src/main/java/org/apache/iceberg/flink/FlinkFilters.java

```diff
@@ -0,0 +1,185 @@
+
+        case IN:
+          List<ResolvedExpression> args = call.getResolvedChildren();
+          FieldReferenceExpression field = (FieldReferenceExpression) args.get(0);
+          List<ResolvedExpression> values = args.subList(1, args.size());
+
+          List<Object> inputValues = values.stream().filter(
+              expression -> {
+                if (expression instanceof ValueLiteralExpression) {
+                  return !((ValueLiteralExpression) flinkExpression).isNull();
+                }
+
+                return false;
+              }
+          ).map(expression -> {
+            ValueLiteralExpression valueLiteralExpression = (ValueLiteralExpression) expression;
+            return valueLiteralExpression.getValueAs(valueLiteralExpression.getOutputDataType().getConversionClass())
+                .get();
+          }).collect(Collectors.toList());
+          return Expressions.in(field.getName(), inputValues);
+
+        case NOT:
+          Expression child = convert(call.getResolvedChildren().get(0));
+          if (child != null) {
+            return Expressions.not(child);
+          }
+
+          return null;
+
+        case AND: {
+          return convertLogicExpression(Expressions::and, call);
+        }
+
+        case OR: {
+          return convertLogicExpression(Expressions::or, call);
+        }
+      }
+    }
+
+    return null;
+  }
+
+  private static Expression convertLogicExpression(BiFunction<Expression, Expression, Expression> function,
+                                                   CallExpression call) {
+    List<ResolvedExpression> args = call.getResolvedChildren();
+    Expression left = convert(args.get(0));
+    Expression right = convert(args.get(1));
+    if (left != null && right != null) {
+      return function.apply(left, right);
+    }
+
+    return null;
+  }
+
+  private static Expression convertComparisonExpression(BiFunction<String, Object, Expression> function,
+                                                        CallExpression call) {
+    List<ResolvedExpression> args = call.getResolvedChildren();
+    FieldReferenceExpression fieldReferenceExpression;
+    ValueLiteralExpression valueLiteralExpression;
+    if (literalOnRight(args)) {
+      fieldReferenceExpression = (FieldReferenceExpression) args.get(0);
+      valueLiteralExpression = (ValueLiteralExpression) args.get(1);
+    } else {
+      fieldReferenceExpression = (FieldReferenceExpression) args.get(1);
+      valueLiteralExpression = (ValueLiteralExpression) args.get(0);
+    }
+
+    String name = fieldReferenceExpression.getName();
+    Class clazz = valueLiteralExpression.getOutputDataType().getConversionClass();
+    Object value = valueLiteralExpression.getValueAs(clazz).get();
+
+    BuiltInFunctionDefinition functionDefinition = (BuiltInFunctionDefinition) call.getFunctionDefinition();
+    if (functionDefinition.equals(BuiltInFunctionDefinitions.EQUALS)) {
```

Comment:
I think we may want to rewrite `NOT_EQUALS` to `notNaN` as well as `notEquals` in Iceberg also doesn't accept NaN as literal; I think SparkFilters doesn't do that because there's no `NotEqualTo` filter in Spark.